### PR TITLE
Relax DNS requirements on shared VPCs

### PR DESCRIPTION
**Pull request by justinsb**
Don't require EnableDNSHostnames on a shared VPC in >= 1.5.0

Create a feature flag for tolerating EnableDNSSupport=false.

Issue #786

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1673)

### CODE REVIEW

File: pkg/model/context.go [hunk 1]
```diff
@@ -190,3 +192,33 @@ func (m *KopsModelContext) UsePrivateDNS() bool {
 
 	return false
 }
+
+// KubernetesVersion parses the semver version of kubernetes, from the cluster spec
+func (c *KopsModelContext) KubernetesVersion() (semver.Version, error) {
```
Comment (chrislovecnm) [inline pkg/model/context.go]: you pulling this out of versions.go?

File: pkg/model/context.go [hunk 2]
```diff
@@ -190,3 +192,33 @@ func (m *KopsModelContext) UsePrivateDNS() bool {
...
+	sv, err := util.ParseKubernetesVersion(kubernetesVersion)
+	if err != nil {
+		return semver.Version{}, fmt.Errorf("unable to determine kubernetes version from %q", kubernetesVersion)
+	}
+
+	return *sv, nil
+}
+
+// VersionGTE is a simplified semver comparison
+func VersionGTE(version semver.Version, major int, minor int) bool {
```
Comment (chrislovecnm) [inline pkg/model/context.go]: Does the semver version not work?

File: pkg/model/context.go (hunk 1 above)
Comment (justinsb) [inline pkg/model/context.go]: This is copy-pasta from https://github.com/kubernetes/kops/blob/master/pkg/model/components/context.go#L32.  I'll add a comment saying to dedup

File: pkg/model/context.go (hunk 2 above)
Comment (justinsb) [inline pkg/model/context.go]: I've seen people make mistakes ;-)

Also, is 1.5.0-alpha1 > or < 1.5.0 ?

File: pkg/model/context.go [hunk 3]
```diff
@@ -190,3 +192,33 @@ func (m *KopsModelContext) UsePrivateDNS() bool {
...
+	if version.Major > major {
+		return true
+	}
+	if version.Major > major {
+		return true
+	}
```
Comment (justinsb) [inline pkg/model/context.go]: Will remove the double-check here.

**Comment by chrislovecnm** [comment]
/lgtm 

Waiting on e2e

**Pull request merged**